src/lib/opts.rs

```diff
@@ -363,20 +366,209 @@ pub fn setup() -> Opts {
+            );
+
+            opts.fastqs
+                .iter()
+                .zip(opts.read_structures.iter())
+                .map(|(f, r)| FastqsAndReadStructure {
+                    fastqs: vec![f.clone()],
+                    read_structure: r.clone(),
+                })
+                .collect()
+        } else if opts.fastqs.iter().all(|f| !f.is_file()) {
+            ensure!(
+                opts.read_structures.is_empty(),
+                "Read Structure must not be given when the input FASTQs are a path prefix."
+            );
+
+            // glob all the FASTQs
+            let input_fastqs: Vec<InputFastq> =
+                opts.fastqs.iter().flat_map(|prefix| InputFastq::glob(&prefix).unwrap()).collect();
+
+            // must find some FASTQs
+            ensure!(!input_fastqs.is_empty(), "No FASTQS found for prefix(es): {:?}.", opts.fastqs);
+
+            // The next few steps (1) group FASTQS by kind and kind number, (2) re-order the groups
+            // by the kind and kind number, and (3) sorts the FASTQS in each group by prefix and
+            // lane.  (1) the group of FASTQS with the same kind and kind number must be read in
+            // serially, and (3) ensures that the FASTQS are synchronized across groups (e.g. the
+            // same lane/prefix across kind/kind-number will be read at the same time).  (2)
+            // ensures that we emit the read pairs in the correct order (e.g. R1/R2).
+
+            // Collect all FASTQs by kind and kind number
+            let mut unsorted_groups: HashMap<(SegmentType, u32), Vec<InputFastq>> = HashMap::new();
+            for f in input_fastqs {
+                let key = (f.kind, f.kind_number);
+                unsorted_groups.entry(key).or_insert_with(|| Vec::with_capacity(1)).push(f);
+            }
+
+            // Sort across each group of FASTQS by kind and kind number.  For each group, sort by
+            // prefix and lane.
+            unsorted_groups
+                .into_iter()
+                .sorted_by(
+                    |((left_kind, left_kind_number), _), ((right_kind, right_kind_number), _)| {
+                        left_kind.cmp(right_kind).then(left_kind_number.cmp(right_kind_number))
+                    },
+                )
+                .map(|((kind, _), fastqs)| {
+                    // this should always succeed
+                    let read_structure =
+                        ReadStructure::from_str(&format!("{}{}", ANY_LENGTH_STR, kind.value()))
+                            .unwrap();
+
+                    // sort FASTQs by prefix and lane for consistent ordering across groups
+                    let fastqs = fastqs
+                        .iter()
+                        .sorted_by(|left, right| {
+                            left.prefix.cmp(&right.prefix).then(left.lane.cmp(&right.lane))
+                        })
+                        .map(|f| f.path.clone())
+                        .collect();
```

Comment:
Since the suffix is fixed (we parsed it), then it is fine to sort by the prefix (then lane).   In fact, we want lane number to be ascending (for the same prefix), so I think if the sort does something weird (L002->L001->L003) we'd get weird results.  Even more, the file pattern of per-tile FASTQs is significantly different than per-lane FASTQs, and there has been no request to support per-tile FASTQs.  Deferring until needed.